src/middlewared/middlewared/plugins/catalog/app_version.py

```diff
@@ -45,32 +45,31 @@ class Config:
         Str('icon_url', required=True, null=True),
         Str('home', required=True),
     ))
-    def get_item_details(self, item_name, options):
+    def get_app_details(self, app_name, options):
         """
-        Retrieve information of `item_name` `item_version_details.catalog` catalog item.
+        Retrieve information of `app_name` `app_version_details.catalog` catalog app.
         """
-        catalog = self.middleware.call_sync('catalog.get_instance', options['catalog'])
-        item_location = os.path.join(catalog['location'], options['train'], item_name)
-        if not os.path.exists(item_location):
-            raise CallError(f'Unable to locate {item_name!r} at {item_location!r}', errno=errno.ENOENT)
-        elif not os.path.isdir(item_location):
-            raise CallError(f'{item_location!r} must be a directory')
+        catalog = self.middleware.call_sync('catalog.config')
+        app_location = os.path.join(get_train_path(catalog['location']), options['train'], app_name)
+        if not os.path.exists(app_location):
```

Comment:
We need to clean this up while you're here. os.path.exists and os.path.isdir stat the underlying object. We can reduce syscalls by donig this all at once.
```
try
    if not stat.S_ISDIR(os.stat(app_location).st_mode):
        raise CallError(f'{app_location!r} must be a directory')
except FileNotFoundError:
    raise CallError(f'Unable to locate {app_name!r} at {app_location!r}', errno=errno.ENOENT)